bigquery_etl/shredder/delete.py

```diff
@@ -334,6 +335,24 @@ class Partition:
     id: Optional[str] = None
     is_special: bool = False
 
+    @property
+    def on_condition(self):
+        """Condition to use in the ON clause of joins.
+
+        Use partition condition when shredding a whole table and TRUE when
+        shredding a single partition.
+        """
+        return self.condition if self.id is None else "TRUE"
+
+    @property
+    def where_condition(self):
```

Comment:
`select_where_condition` might be a slightly more accurate name.  At first I thought maybe `partition.condition` still being used in the `DELETE` statement was an oversight, but then I came back and read the description here more carefully (kudos for having docstrings).